pkg/encoder/token_serializer.go

```diff
@@ -0,0 +1,39 @@
+package encoder
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/openfga/openfga/pkg/storage"
+)
+
+type ContinuationTokenSerializer interface {
+	// Serialize serializes the continuation token into a format readable by ReadChanges
+	Serialize(ulid string, objType string) (token []byte, err error)
+
+	// Deserialize deserializes the continuation token into a format readable by ReadChanges
+	Deserialize(token string) (ulid string, objType string, err error)
```

Comment:
If in the future you want to introduce the same artifact for the ReadPage API or ListObjects API, what will you have to change? You will have to create a new interface or add new methods. The existing interface name makes it sound like it will work for every API (but it won't), and new methods will have to be called `DeserializeForListObjects` to distinguish them from the existing methods. 

I would have done this like:

```go
type ReadChangesContinuationTokenSerializer interface {
   Seralize(...)
   Deserialize(..)
}
```

or 

```go
type ContinuationTokenSerializer interface {
   SeralizeForReadChanges(...)
   DeserializeForReadChanges(..)
}
```

Reply:
This change was made based on [Jose's comment](https://github.com/openfga/openfga/pull/2020#discussion_r1821402533)